lib/common/xml.c

```diff
@@ -2955,11 +2955,9 @@ pcmk__xe_set_propv(xmlNodePtr node, va_list pairs)
         }
 
         value = va_arg(pairs, const char *);
-        if (value == NULL) {
-            return;
+        if (value != NULL) {
```

Comment:
How about using `pcmk__str_empty` here instead?

Reply:
An empty string is a legitimate value that can be set. I don't know of anywhere off the top of my head where that would be significant, but I'd be wary of closing that possibility.

There's a subtle catch in our current design for adding XML attributes. We have two primary functions, crm_xml_add() which skips NULL values (meaning any already-existing value will be left intact), and crm_xml_replace() which deletes any attribute passed with a NULL value. Eventually I'd like to come up with a design that makes that more explicit (whether clearer function names, or one function that takes a flag for it).

That doesn't matter in the original situation here since we're creating a new XML node and there are no existing attribute values. But I think it's relevant for how pcmk__xe_set_propv() should handle NULL vs empty string -- since it follows crm_xml_add() for NULLs, I think it makes sense to follow crm_xml_add() for empty strings as well.